tools/deps_info.py

```diff
@@ -105,52 +105,49 @@
   'emscripten_pc_get_function': ['malloc', 'free'],
   'emscripten_run_preload_plugins_data': ['malloc'],
   'emscripten_run_script_string': ['emscripten_builtin_malloc', 'emscripten_builtin_free'],
-  'emscripten_set_batterychargingchange_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_batterylevelchange_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_beforeunload_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_blur_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_click_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_dblclick_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_devicemotion_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_deviceorientation_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_focus_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_focusin_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_focusout_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_fullscreenchange_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_gamepadconnected_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_gamepaddisconnected_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_keydown_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_keypress_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_keyup_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_mousedown_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_mouseenter_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_mouseleave_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_mousemove_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_mouseout_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_mouseover_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_mouseup_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_offscreencanvas_size_on_target_thread_js': ['malloc', 'free'],
-  'emscripten_set_orientationchange_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_pointerlockchange_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_pointerlockerror_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_resize_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_scroll_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_touchcancel_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_touchend_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_touchmove_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_touchstart_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_visibilitychange_callback_on_thread': ['malloc', 'free'],
-  'emscripten_set_wheel_callback_on_thread': ['malloc', 'free'],
+  'emscripten_set_batterychargingchange_callback_on_thread': ['malloc'],
```

Comment:
I'm having trouble seeing `free` is no longer needed by these symbols?  (but malloc is?)

Reply:
I think the removal of `libraries.push('library_webgl.js');` when building with `-sUSE_PTHREADS` made this JS dependency disappear. See:
https://github.com/emscripten-core/emscripten/blob/a2d0b4e533d1af75e0d2897c805cea0b7572799a/tests/test_other.py#L11023-L11024

```bash
$ grep -nH "_free(" src/library_pthread.js
$ grep -nH "_free(" src/library_html5.js
$ grep -nH "_free(" src/library_html5_webgl.js
$ grep -nH "_free(" src/library_webgl.js
src/library_webgl.js:1071:      _free(GL.contexts[contextHandle].handle);
src/library_webgl.js:4117:    _free(mapping.mem);
```